api/docker/entrypoint.sh

```diff
@@ -8,8 +8,26 @@ if [[ "${MIGRATION_ENABLED}" == "true" ]]; then
 fi
 
 if [[ "${MODE}" == "worker" ]]; then
-  exec celery -A app.celery worker -P ${CELERY_WORKER_CLASS:-gevent} -c ${CELERY_WORKER_AMOUNT:-1} --loglevel INFO \
+
+  # Get the number of available CPU cores
+  AVAILABLE_CORES=$(nproc)
+  if [ "${CELERY_AUTO_SCALE,,}" = "true" ]; then
+    # Set MAX_WORKERS to the number of available cores if not specified
+    MAX_WORKERS=${CELERY_MAX_WORKERS:-$AVAILABLE_CORES}
+    MIN_WORKERS=${CELERY_MIN_WORKERS:-1}
+    CONCURRENCY_OPTION="--autoscale=${MAX_WORKERS},${MIN_WORKERS}"
+  elif [ -n "$CELERY_WORKER_AMOUNT" ]; then
+    # If CELERY_WORKER_AMOUNT is set, use it
+    CONCURRENCY_OPTION="-c ${CELERY_WORKER_AMOUNT}"
+  else
+    # Otherwise, the default is the number of CPUs available on the system
+    CONCURRENCY_OPTION=""
```

Comment:
I think it should be set to 1 even if CELERY_WORKER_AMOUNT is not set, otherwise users will automatically start multiple workers after upgrading to the next version, resulting in insufficient memory resources and OOM.